user_app/bin/menu/base/entry.dart

```diff
@@ -0,0 +1,49 @@
+import 'dart:collection';
+
+import 'multi_value_field.dart';
+import 'menu.dart';
+import 'field.dart';
+import 'single_value_field.dart';
+
+/// An entry is the type of menu that accepts values.
+/// It has a [Map] of [Field] that contains [SingleValueField] or [MultiValueField]
+/// It accepts till all the values of [Field] are accepted.
+
+class Entry extends Menu {
+  Entry(String name) : super(name);
+
+  final Map<String, Field> fields = {};
+
+  void addSingleValueField<T>(
+      String name,
+      String message,
+      T? Function(String? text) cast,
+      bool Function(T?)? validationFunction) {
```

Comment:
Start using named parameters. Dart supports named parameters, its easier to call methods with named parameters and intellisense

Reply:
Okay